app/javascript/components/AeInlineMethod/NamespaceSelector.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { Loading } from 'carbon-components-react';
+import FilterNamespace from './FilterNamespace';
+import MiqDataTable from '../miq-data-table';
+import NotificationMessage from '../notification-message';
+import { CellAction } from '../miq-data-table/helper';
+import {
+  methodSelectorHeaders, formatMethods, searchUrl, namespaceUrls,
+} from './helper';
+import './style.scss';
+
+/** Component to search and select AeMethods. */
+const NamespaceSelector = ({ onSelectMethod, selectedIds }) => {
+  const [data, setData] = useState({
+    domains: [],
+    methods: [],
+    loading: true,
+    searchText: undefined,
+    selectedDomain: undefined,
+  });
+
+  /** Loads the 'domains' and 'methods' from its respective URL's during the component's onLoad event. */
+  useEffect(() => {
+    Promise.all([
+      http.get(namespaceUrls.aeDomainsUrl),
+      http.get(namespaceUrls.aeMethodsUrl)])
+      .then(([{ domains }, { methods }]) => {
+        setData({
+          ...data,
+          domains,
+          loading: false,
+          methods: formatMethods(methods),
+        });
+      });
+  }, []);
+
+  /** Function to handle search text and drop-down item onchange events. */
+  const handleFilterOnChange = (filterData) => {
+    const text = filterData.text ? filterData.text : data.searchText;
+    const selectedDomain = filterData.selectedDomain ? filterData.selectedDomain : data.selectedDomain;
+    const url = searchUrl(selectedDomain, text);
+    http.get(url)
```

Comment:
if the load time is huge, then maybe we can avoid this URL which run on the filter's change event, and filter the existing records that were once loaded and available in `data.methods`

Reply:
TODO: need to see if we can make use of the `useCallbacks` react hook.